tutorials/strings/manachers-algorithm.md

```diff
@@ -9,4 +9,174 @@ keywords:
   - algorithm
 ---
 
-Coming Soon
\ No newline at end of file
+
+# Manacher's Algorithm
+
+There are many ways to find the `longest palindromic substring`. One can find it in O(N³) and also in O(N²) time complexity. But using `Manacher's Algorithm` we do it in `O(N) time`. This algorithm was discovered by `Glenn K. Manacher` in 1975.
+
+Given a string, find the longest substring which is palindrome. 
+
+* if the given string is “abaaba”, the output should be “abaaba”
+* if the given string is “abababa”, the output should be “abababa”
+* if the given string is “abcbabcbabcba”, the output should be “abcbabcbabcba”
```

Comment:
1. don't copy from GFG. 
2. give a brief context on what palindrome is
3. besides you should include one that the longest palindromic substring is not same as the original string. i.e. the one you deleted from that site.